### PR TITLE
Improved code quality

### PR DESCRIPTION
**Pull request by Nityanand13**

### CODE REVIEW

**Comment by Kshitij-Katiyar** [comment]
Give a better title than review fixes done this is PR title not a commit message.Include Issue and PR URL in description

**Review by Kshitij-Katiyar — APPROVED** [review]

File: webapp/src/selectors/index.ts [hunk 1]
```diff
@@ -13,21 +13,14 @@ const getPluginState = (state) => state['plugins-' + PluginId] || {};
 
 export const getPluginServerRoute = (state) => {
     const config = getConfig(state);
-    let basePath = '';
-    let url;
+    let baseUrl = config?.configSiteURL ? config.SiteUrl : window.location.href;
+    let basePath = new URL(baseUrl).pathname;
```
Comment (manojmalik20) [inline webapp/src/selectors/index.ts:17]:
```suggestion
    const baseUrl = new URL(config?.SiteURL ? config.SiteUrl : window.location.href);
    let basePath = baseUrl.pathname;
```

File: webapp/src/selectors/index.ts [hunk 2]
```diff
@@ -13,21 +13,14 @@ const getPluginState = (state) => state['plugins-' + PluginId] || {};
...
-        url = new URL(window.location.href);
-    }
-
-    basePath = url.pathname;
     if (basePath && basePath[basePath.length - 1] === '/') {
         basePath = basePath.substr(0, basePath.length - 1);
     }
 
-    return `${url.origin}${basePath}/plugins/${PluginId}`;
+    return `${new URL(baseUrl).origin}${basePath}/plugins/${PluginId}`;
```
Comment (manojmalik20) [inline webapp/src/selectors/index.ts:23]:
```suggestion
    return `${baseUrl.origin}${basePath}/plugins/${PluginId}`;
```

File: webapp/src/selectors/index.ts (hunk 1 above)
Comment (Nityanand13) [reply webapp/src/selectors/index.ts:17]: I think it should be `const baseUrl = new URL(config?.SiteURL ? config.SiteURL : window.location.href);`

Comment (manojmalik20) [reply webapp/src/selectors/index.ts]: How about
```
const baseUrl = new URL(config?.SiteURL ?? window.location.href);
```

Comment (Nityanand13) [reply webapp/src/selectors/index.ts]: Its even better

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**